src/AutoRest.CSharp/Common/Output/Models/RestClientBuilder.cs

```diff
@@ -599,39 +566,15 @@ public virtual Parameter BuildConstructorParameter(RequestParameter requestParam
                 : new Parameter(name, description, type, null, parameter.Validation, null, RequestLocation: location);
         }
 
-        protected static bool IsMethodParameter(RequestParameter requestParameter)
-            => requestParameter.Implementation == ImplementationLocation.Method && requestParameter.Schema is not ConstantSchema && !requestParameter.IsFlattened && requestParameter.GroupedBy == null;
+        public static bool IsIgnoredHeaderParameter(InputParameter operationParameter)
+            => operationParameter.Location == RequestLocation.Header && IgnoredRequestHeader.Contains(operationParameter.NameInRequest);
 
-        public static bool IsEndpointParameter(RequestParameter requestParameter)
-            => requestParameter.Origin == "modelerfour:synthesized/host";
-
-        public static bool IsContentTypeParameter(RequestParameter requestParameter)
-            => requestParameter.Origin == "modelerfour:synthesized/content-type";
-
-        public static bool IsIgnoredHeaderParameter(RequestParameter requestParameter)
-            => requestParameter.In == HttpParameterIn.Header && IgnoredRequestHeader.Contains(GetRequestParameterName(requestParameter));
-
-        private static bool IsRequestConditionHeader(RequestParameter requestParameter, out RequestConditionHeaders header)
+        private Parameter BuildParameter(in InputParameter operationParameter, Type? typeOverride = null)
         {
-            header = RequestConditionHeaders.None;
-            return requestParameter.In == HttpParameterIn.Header && ConditionRequestHeader.TryGetValue(GetRequestParameterName(requestParameter), out header);
+            CSharpType type = typeOverride != null ? new CSharpType(typeOverride, operationParameter.Type.IsNullable) : _typeFactory.CreateType(operationParameter.Type);
+            return Parameter.FromRequestParameter(operationParameter, type, _typeFactory);
         }
```

Comment:
Is this the trick for [this](https://github.com/Azure/autorest.csharp/pull/2424#discussion_r929682278)?
The only place that previously invoking the CreateType overload with the outsourcing extensions is this `CreateParameter` method. Since this is now taking a InputParameter as parameter instead of RequestParameter from the code model, the extra extension parameter is no longer needed.